rails-commenting.rb

```diff
@@ -7,56 +7,59 @@
     if @post.valid?
       redirect_to blog_post_path(@post)
     end
   end
 
   def edit
     # ---6)
-    @post = BlogPost.find(params[:id])
+    @post = BlogPost.find(params[:id]) # the edit method is similar to the new method in the aspect that it does not actually perform a specific action, it instead creates a form for the user to see so that the update method can perform the create action. this one is different from new being that since the form already exists it needs a param of the ID so that it knows which one you are trying to edit, the edit request will typically be specifically on the blog post or whatever is in your database. Edit would fall in the U of the CRUD acronym for update, I think kinda like new it would also fall under Read since you arent actually performing an update action. its route is and http verbs are put/patch 'reseources/:id' => 'resources#edit' 
   end
 
   def update
     @post = BlogPost.find(params[:id])
     # ---7)
-    @post.update(blog_post_params)
+    @post.update(blog_post_params) # update method is similar to the create method as it actually performs the action of updating the edit form created. Like the create method, this one also requires strong params so that the user or any user cannot just put whatever they want(for security reasons mostly) this method also uses the built in method .find so that it finds the specific instance you are trying to update. and then you are doing the same thing with the if statement for the convenience of the user to see if the update was successful, if the update post request was successful, redirect the user to the blog post path so that they can see the changes made, this one does not redirect to the home page, it is a better experience if you redirect the user the show the blog you actually changed so that they see it instantly without having to click around to get back to the updated page. It falls in the U for update in the CRUD acronym and https verbs are put/patch, route being put/patch 'resources/:id' => 'resources#update' 
     if @post.valid?
       redirect_to blog_post_path(@post)
     end
   end
 
   def destroy
-    @post = BlogPost.find(params[:id])
+    @post = BlogPost.find(params[:id]) #the destroy method is pretty simple. it requires the param of the ID that it is trying to delete, and is responsible for deleting that instance. It does exactly what it says it is going to do, it will give the user the ability to delete that section from the model. In CRUD it is the D for delete, and HTTP verb is also, much to your surprise, delete.. and its route is delete '/resources/:id' => 'resources#destroy' you also want to give a good user experience like the other major actions, so if the post is successfuly deleted redirected them to the main path where they can see that it was deleted from the list of all of the posts. 
     if @post.destroy
       # ---8)
```

Comment:
Exactly, and this one was referencing just that redirect after a post was destroyed.